packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/VirtualizedTraceView.tsx

```diff
@@ -158,6 +159,10 @@ export class VirtualizedTraceViewImpl extends React.PureComponent<VirtualizedTra
     setTrace(trace, uiFind);
   }
 
+  componentDidMount() {
+    setTimeout(this.props.scrollToFirstVisibleSpan);
```

Comment:
Without the `setTimeout` it scrolls to the bottom. I think the rows may not have height in the initial render.
@tiffon do you have any insight on why this would scroll too far without the `setTimeout`

Reply:
Looks like the constructor calls `setTrace(...)`. When visiting the URL, directly, the `setTrace(...)` applies the find matching.

When coming from the search page the `setTimeout` has no effect; the find is already applied to the redux state via the reducer that changed the state to be viewing the trace.

So, seems like the rows are rendered but they're not collapsed, yet, when visiting a trace directly from a URL. Probably it's scrolling to the position of the row, but then the non-matching rows are collapsed and the target scroll position ends up getting truncated with the result of being scrolled to the bottom.